frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -370,6 +376,50 @@ function ReaderDictionary:getNumberOfDictionaries()
     return nb_available, nb_enabled, nb_disabled
 end
 
+function ReaderDictionary:_genDownloadDictionariesMenu()
+    local downloadable_dicts = require("ui/data/dictionaries")
+    local languages = {}
+
+    for i = 1, #downloadable_dicts do
+        local dict = downloadable_dicts[i]
+        local dict_lang_in = dict.lang_in
+        local dict_lang_out = dict.lang_out
+        if not languages[dict_lang_in] then
+            languages[dict_lang_in] = {}
+        end
+        table.insert(languages[dict_lang_in], dict)
+        if not languages[dict_lang_out] then
+            languages[dict_lang_out] = {}
+        end
+        table.insert(languages[dict_lang_out], dict)
+    end
+
+    -- remove duplicates
+    for lang_key,lang in pairs(languages) do
+        local hash = {}
+        local res = {}
+        for k,v in ipairs(lang) do
+           if not hash[v.name] then
+               res[#res+1] = v
```

Comment:
Not used to see this done that way in current code :) We're most often using `table.insert(res, v)`